Xamarin.PropertyEditing.Windows/StringEditorControl.cs

```diff
@@ -1,7 +1,43 @@
-﻿namespace Xamarin.PropertyEditing.Windows
+using System;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace Xamarin.PropertyEditing.Windows
 {
+	[TemplatePart (Name = "TextBox", Type = typeof(TextBox))]
 	public class StringEditorControl
 		: PropertyEditorControl
 	{
+		public override void OnApplyTemplate ()
+		{
+			base.OnApplyTemplate ();
+
+			this.textBox = GetTemplateChild("TextBox") as TextBox;
+			if (this.textBox == null)
+				throw new InvalidOperationException ("StringEditorControl must have a 'TextBox' TextBox part");
+
+			this.textBox.PreviewMouseLeftButtonDown += OnLeftDown;
+			this.textBox.GotKeyboardFocus += OnGotFocus;
+			this.textBox.MouseDoubleClick += OnGotFocus;
+		}
+
+		private TextBox textBox;
+
+		private void OnGotFocus (object sender, EventArgs e)
```

Comment:
nit: not a fan of the handler's name here, since it's used on double click as well, and it doesn't convey what it does either.